### PR TITLE
chore: fix multirm unit test flake

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Description
require error.Contains for multiRM unit test that returns a list of resource managers. This way, we'll avoid failing the test if the list is a different order.

## Test Plan
Pass test

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/rm/multirm/multirm_intg_test.go [hunk 1]
```diff
@@ -733,7 +734,15 @@ func TestGetRMName(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			rmName, err := mockMultiRM.getRM(tt.rmName, tt.rpName)
 			require.Equal(t, tt.expectedRMName, rmName)
-			require.Equal(t, tt.err, err)
+			if tt.err != nil && strings.Contains(tt.err.Error(), "exists for both resource managers") {
+				// Manually checking for ErrRMConflict because RMs may be returned in different order.
+				require.ErrorContains(t, err, "exists for both resource managers")
+				require.ErrorContains(t, err, "aws")
+				require.ErrorContains(t, err, "gcp")
```
Comment (stoksc) [inline master/internal/rm/multirm/multirm_intg_test.go:741]: weird to hard code it this way to me. should it just use parameters in the test case instead?

Comment (carolinaecalderon) [reply master/internal/rm/multirm/multirm_intg_test.go:741]: I could introduce an extra parameter to use just in this case.
is that preferable? even if the other 7/8 cases won't need it ?

File: master/internal/rm/multirm/multirm_intg_test.go [hunk 2]
```diff
@@ -719,21 +720,33 @@ func TestGetRMName(t *testing.T) {
...
+		},
+		{"RM undefined, RP defined/doesn't exist", "", "gcp3", ErrRPNotDefined("gcp3"), "", nil},
 	}
 	for _, tt := range cases {
 		t.Run(tt.name, func(t *testing.T) {
 			rmName, err := mockMultiRM.getRM(tt.rmName, tt.rpName)
 			require.Equal(t, tt.expectedRMName, rmName)
-			require.Equal(t, tt.err, err)
+			if tt.err != nil && strings.Contains(tt.err.Error(), "exists for both resource managers") {
+				require.ErrorContains(t, err, "exists for both resource managers")
```
Comment (jesse-amano-hpe) [inline master/internal/rm/multirm/multirm_intg_test.go:743]: Might be good to check error string length here too, to make sure the conflict checking behavior doesn't miss or over-count one.
```suggestion
				require.ErrorContains(t, err, "exists for both resource managers")
				require.Equal(t, len(tt.err.Error()), len(err.Error())
```

File: master/internal/rm/multirm/multirm_intg_test.go [hunk 3]
```diff
@@ -719,21 +720,33 @@ func TestGetRMName(t *testing.T) {
 		rpName         string
 		err            error
 		expectedRMName string
+		rmConflicts    []string
```
Comment (NicholasBlaskey) [inline master/internal/rm/multirm/multirm_intg_test.go:723]: I have a different take on this.

Let's just sort rmNames here https://github.com/determined-ai/determined/blob/e8b0165413fbe1142eea29de56fd860e698f8248/master/internal/rm/multirm/multirm.go#L22

I also like think generally having one canonical error message for the same issue is better than being able to return a bunch

Comment (carolinaecalderon) [reply master/internal/rm/multirm/multirm_intg_test.go]: agree that sorting on rmNames is the best option -- pushed a new commit with that

**Review by NicholasBlaskey — APPROVED** [review]

**Pull request merged**